dist/prototypes.py

```diff
@@ -4,55 +4,72 @@
-        l = p.strip()
+def clean_function_name(filename, fn):
+    ret = fn.strip()
 
-        # Ignore statics in XXX.c files.
-        if fnmatch.fnmatch(name, "*.c") and 'static' in l:
-                continue
+    # Ignore statics in XXX.c files.
+    if fnmatch.fnmatch(filename, "*.c") and 'static' in ret:
+        return None
 
-        # Join the first two lines, type and function name.
-        l = l.replace("\n", " ", 1)
+    # Join the first two lines, type and function name.
+    ret = ret.replace("\n", " ", 1)
 
-        # If there's no CPP syntax, join everything.
-        if not '#endif' in l:
-            l = " ".join(l.split())
+    # If there's no CPP syntax, join everything.
+    if not '#endif' in ret:
+        ret = " ".join(ret.split())
 
-        # If it's not an inline function, prefix with "extern".
-        if not 'inline' in l:
-                l = 'extern ' + l
+    # If it's not an inline function, prefix with "extern".
+    if not 'inline' in ret:
+        ret = 'extern ' + ret
 
-        # Switch to the include file version of any gcc attributes.
-        l = l.replace("WT_GCC_FUNC_ATTRIBUTE", "WT_GCC_FUNC_DECL_ATTRIBUTE")
+    # Switch to the include file version of any gcc attributes.
+    ret = ret.replace("WT_GCC_FUNC_ATTRIBUTE", "WT_GCC_FUNC_DECL_ATTRIBUTE")
 
-        # Everything but void requires using any return value.
-        if not re.match(r'(static inline|extern) void', l):
-                l = l + " WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result))"
+    # Everything but void requires using any return value.
+    if not re.match(r'(static inline|extern) void', ret):
+        ret = ret + " WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result))"
 
-        # If a line ends in #endif, appending a semicolon results in an illegal
-        # expression, force an appended newline.
-        if re.match(r'#endif$', l):
-            l = l + '\n'
+    # If a line ends in #endif, appending a semicolon results in an illegal
+    # expression, force an appended newline.
+    if re.match(r'#endif$', ret):
+        ret = ret + '\n'
+
+    return ret + ';\n'
+
+# Build function prototypes from a list of files.
+def fn_prototypes(fns, tests, name):
+    s = open(name, 'r').read()
+    for p in re.findall(r'\n[A-Za-z_].*\n__wt_[^{]*', s):
+        clean = clean_function_name(name, p)
+        if clean is not None:
+            fns.append(clean)
 
-        # Add the trailing semi-colon.
-        list.append(l + ';\n')
+    s = open(name, 'r').read()
+    for p in re.findall(r'\n[A-Za-z_].*\n__ut_[^{]*', s):
+        clean = clean_function_name(name, p)
+        if clean is not None:
+            tests.append(clean)
 
 # Write results and compare to the current file.
-def output(p, f):
+def output(fns, tests, f):
     tmp_file = '__tmp'
     tfile = open(tmp_file, 'w')
-    for e in sorted(list(set(p))):
+    for e in sorted(list(set(fns))):
         tfile.write(e)
+
+    tfile.write('\n#ifdef HAVE_UNITTEST\n')
```

Comment:
The additional processing of the unit tests looks good. I'd suggest some comments at the top of the file explaining how the processing of both the wt and the ut functions happens and why so that future readers of the code will find it clear.

Reply:
Yep, that's a good point. I'll address this and your previous comment once I've made the evergreen linter happy.

Reply:
Fixed in 6e8ada5